tests/src/Functional/OgSelectionWidgetOptionsTest.php

```diff
@@ -0,0 +1,159 @@
+   *
+   * @var \Drupal\user\Entity\User
+   */
+  protected $groupAdministratorUser;
+
+  /**
+   * A non-member user.
+   *
+   * @var \Drupal\user\Entity\User
+   */
+  protected $nonMemberUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Create group node types.
+    NodeType::create(['type' => 'group1'])->save();
+    NodeType::create(['type' => 'group2'])->save();
+    Og::addGroup('node', 'group1');
+    Og::addGroup('node', 'group2');
+
+    NodeType::create(['type' => 'group_content'])->save();
+
+    // Use a select list widget for the audience field, so it's easier to get
+    // all the values.
+    $settings = [
+      'form_display' => [
+        'type' => 'options_select',
+      ],
+    ];
+    Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, 'node', 'group_content', $settings);
+
+    // Create users.
+    $this->groupMemberUser = $this->drupalCreateUser();
+    $this->groupOwnerUser = $this->drupalCreateUser();
+    $this->groupAdministratorUser = $this->drupalCreateUser(['administer group']);
+    $this->nonMemberUser = $this->drupalCreateUser();
+
+    // Create groups.
+    $this->group1 = Node::create([
+      'type' => 'group1',
+      'title' => 'group1',
+      'uid' => $this->groupOwnerUser->id(),
+    ]);
+    $this->group1->save();
+
+    $this->group2 = Node::create([
+      'type' => 'group2',
+      'title' => 'group2',
+      'uid' => $this->groupOwnerUser->id(),
+    ]);
+    $this->group2->save();
+
+    // @todo: Add unpublished node.
+
+    // Add member to group.
+    Og::createMembership($this->group1, $this->groupMemberUser)->save();
+    Og::createMembership($this->group2, $this->groupMemberUser)->save();
+  }
+
+  /**
+   * Tests the group audience widgets shows correct values.
+   */
+  public function testNonRequiredAudienceField() {
+    // Non member user.
+    $this->drupalLogin($this->nonMemberUser);
+    $this->drupalGet('node/add/group_content');
+    $this->assertSession()->statusCodeEquals(403);
+
+    // Group member without create permissions.
+    $this->drupalLogin($this->groupMemberUser);
+    $this->drupalGet('node/add/group_content');
+    $this->assertSession()->statusCodeEquals(403);
+
+    // Grant create permission for the first group.
+    $role = OgRole::getRole($this->group1->getEntityTypeId(), $this->group1->bundle(), OgRoleInterface::AUTHENTICATED);
+    $role
+      ->grantPermission('create node group_content')
+      ->save();
+
+    $this->drupalGet('node/add/group_content');
+    $this->assertSession()->optionExists('Groups audience', '_none');
+    $this->assertSession()->optionExists('Groups audience', $this->group1->label());
+    $this->assertSession()->optionNotExists('Groups audience', $this->group2->label());
+
+    // Group owner.
+    $this->drupalLogin($this->groupOwnerUser);
+    $this->drupalGet('node/add/group_content');
+    $this->assertSession()->optionExists('Groups audience', $this->group1->label());
```

Comment:
Yeah, we have some more work here. As a non-admin user, when creating new group content I'm getting notices. It's related to this - https://github.com/Gizra/og/pull/181/files#diff-669ced81d1fae6b8b9a940fcf8dc8d35R91 , so I'm going to dig in..